### PR TITLE
test: update the sim tests to simplify assertion logic

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Make the sim tests more readable and easy to maintain. 

**Description**

- Updated the assertion logic to make it flexible and more readable. 
- Updated the matcher to be applied for both capture and assert step using bit flag
- Move the capture step based on the matcher not every slot
- Move the assert step to be applied once every node, not on all nodes at once. 
- Fixed the head assertion failure 
- Updated the error logging to make it more consistent among assertions  

**Steps to test or reproduce**

Run all tests

### CODE REVIEW

File: .github/workflows/test-sim.yml [hunk 1]
```diff
@@ -6,6 +6,11 @@ on:
     branches: [unstable, stable]
   pull_request:
   workflow_dispatch:
+    inputs:
+      debug:
```
Comment (matthewkeil) [inline .github/workflows/test-sim.yml:10]: What `type` of input will this render as?  Does it default to `string` with the `default: ''`?

File: packages/cli/test/utils.ts [hunk 2]
```diff
@@ -38,6 +38,6 @@ export function getTestdirPath(filepath: string): string {
   return fullpath;
 }
 
-export function isTruthy<T = unknown>(value: T | undefined | null): value is T {
-  return value !== undefined && value !== null;
+export function isNullish<T = unknown>(value: T | undefined | null): value is undefined | null {
+  return value === undefined || value === null;
```
Comment (matthewkeil) [inline packages/cli/test/utils.ts:42]: This is a nice change.  Is more readable 🦅 👁️

**Review by matthewkeil — APPROVED** [review]
Nice job.  I reviewed for code quality.  Only had the one question about workflow input.  Approving as is

File: .github/workflows/test-sim.yml (hunk 1 above)
Comment (nazarhussain) [reply .github/workflows/test-sim.yml:10]: If you invoke the workflow manually you can set any value to it e.g. `lodestar:sim:*` will show all sim tests related logs. For regular run which is invoked with push or pull request, it's value will not be defined.

Comment (matthewkeil) [reply .github/workflows/test-sim.yml:10]: sweet.  it defaults to a `type: string` control

Screen Shot 2023-05-22 at 11 41 46 AM

**Comment by dapplion** [comment]
Looks good!

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.9.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.9.0) :tada:

File: packages/cli/test/utils/simulation/SimulationTracker.ts [hunk 3]
```diff
@@ -279,52 +270,115 @@ export class SimulationTracker {
...
-      }
     }
 
-    this.reporter.progress(slot);
-    this.processRemoveAssertionQueue();
-  }
-
-  private processRemoveAssertionQueue(): void {
-    for (const id of this.removeAssertionQueue) {
+    for (const id of removeAssertions) {
```
Comment (nflaig) [inline packages/cli/test/utils/simulation/SimulationTracker.ts:375]: @nazarhussain Noticed that removeAssertions is always empty here as there are no items pushed to array after initializing it

https://github.com/ChainSafe/lodestar/blob/29314c9e4515662fd882b9a47b8aa0c95b0ff88c/packages/cli/test/utils/simulation/SimulationTracker.ts#L330

**Pull request merged**